tests/jballerina-unit-test/src/test/resources/test-src/types/constant/simple-literal-constant-negative.bal

```diff
@@ -321,3 +321,35 @@ Ints2 ints2 = 2;
 
 float:NaN floatNan = 1.0;
 float:Infinity floatInf = 1.0;
+
+// Test arithmatic errors
+const NUM1 = -1;
+const int NUM2 = -9223372036854775807 - 1;
+const int NUM3 = 0;
+const int NUM4 = 9223372036854775807;
+const int NUM5 = 10;
+
+const int ANS1 = NUM2 / NUM1;          // error
+const int ANS2 = NUM2 / NUM3;          // error
+
+const int ANS3 = NUM2 * NUM1;          // error
+const int ANS4 = NUM2 * 5;             // error
+
+const int ANS5 = NUM2 % NUM3;          // error
+const int ANS6 = -NUM2;                // error
+
+const int ANS7 = NUM2 + NUM1;          // error
+const int ANS8 = NUM4 + NUM5;          // error
+
+const int ANS10 = NUM2 - NUM5;         // error
+const int ANS11 = NUM4 - NUM1;         // error
+
+const decimal ANS12 = 5 % 0;           // error
+
+const int ANS7 = NUM2 - 1;             // error
+
+const decimal d1 = 9.999999999999999999999999999999999E6001d * 1E145d;   // error
+const decimal d2 = -9.999999999999999999999999999999999E6141d * 1E5d;    // error
+const decimal d3 = 9.999999999999999999999999999999999E6144d + 1E6143d;  // error
+const decimal d4 = -1E6144d - 9.999999999999999999999999999999999E6144d; // error
+const decimal d5 = 1E614d / 2E-5800d;                                    // error
```

Comment:
Shall we add some tests for unary operators as well?

Reply:
Added 9d9f49229743a0d2c9229b098cc2bde7a8d31c25

There is an existing test as well https://github.com/ballerina-platform/ballerina-lang/pull/37726/files#diff-f7b80acafb6a5c9b42be1aea1acfa78b59911c59f6a479ac5cc5265b47e1a213R339